sdks/java/io/kinesis/src/test/java/org/apache/beam/sdk/io/kinesis/KinesisReaderTest.java

```diff
@@ -119,4 +127,85 @@ public void readsThroughAllDataAvailable() throws IOException, TransientKinesisE
     assertThat(reader.advance()).isFalse();
   }
 
+  @Test
+  public void watermarkDoesNotChangeWhenToFewSampleRecords()
+      throws IOException, TransientKinesisException {
+    final long timestampMs = 1000L;
+
+    KinesisRecord firstRecord = prepareRecordMockWithArrivalTimestamp(timestampMs);
```

Comment:
Seems like lines 135-142 can be abstracted into a nice helper method:
prepareRecordsWithArrivalTimestamps(timestampMs, 1 /* increment */, 5 /* count */);

and it can be reused in the 2 other tests